test/validation/api/classification/odp_classification_common.c

```diff
@@ -91,98 +91,58 @@ int stop_pktio(odp_pktio_t pktio)
-	uint32_t hlen = 0;
 	int status;
 
 	data.magic = DATA_MAGIC;
 	data.seq = ++seq;
 
-	ip = (odph_ipv4hdr_t *)odp_packet_l3_ptr(pkt, NULL);
-	offset = odp_packet_l4_offset(pkt);
-	CU_ASSERT_FATAL(offset != ODP_PACKET_OFFSET_INVALID);
-
-	if (ip->proto == ODPH_IPPROTO_IGMP) {
-		status = odp_packet_copy_from_mem(pkt, offset + ODP_IGMP_HLEN,
-						  sizeof(data), &data);
-	} else if (ip->proto == ODPH_IPPROTO_ICMPV4) {
-		status = odp_packet_copy_from_mem(pkt, offset + ODPH_ICMPHDR_LEN,
-						  sizeof(data), &data);
-	} else if (ip->proto == ODPH_IPPROTO_SCTP) {
-		/* Create some invalid SCTP packet for testing under the assumption that
-		 * no implementation really cares
-		 */
-		status = odp_packet_copy_from_mem(pkt, offset + ODPH_SCTPHDR_LEN,
-						  sizeof(data), &data);
-	} else if (ip->proto == ODPH_IPPROTO_UDP) {
-		udp = (odph_udphdr_t *)odp_packet_l4_ptr(pkt, NULL);
-		port = odp_be_to_cpu_16(udp->dst_port);
-		if (port == ODP_GTPU_UDP_PORT) {
-			hlen = offset + ODPH_UDPHDR_LEN + ODP_GTP_HLEN;
-			status = odp_packet_copy_from_mem(pkt, hlen,
-							  sizeof(data), &data);
-		} else {
-			status = odp_packet_copy_from_mem(pkt, offset + ODPH_UDPHDR_LEN,
-							  sizeof(data), &data);
-		}
-	} else {
-		tcp = (odph_tcphdr_t *)odp_packet_l4_ptr(pkt, NULL);
-		status = odp_packet_copy_from_mem(pkt, offset + tcp->hl * 4,
-						  sizeof(data), &data);
-	}
+	offset = seqno_offset(pkt);
+
+	status = odp_packet_copy_from_mem(pkt, offset, sizeof(data), &data);
 
 	return status;
 }
 
 uint32_t cls_pkt_get_seq(odp_packet_t pkt)
 {
-	uint32_t offset;
 	cls_test_packet_t data;
-	odph_ipv4hdr_t *ip;
-	odph_tcphdr_t *tcp;
-	odph_udphdr_t *udp;
-	uint32_t hlen = 0;
-	uint16_t port = 0;
-
-	ip = (odph_ipv4hdr_t *)odp_packet_l3_ptr(pkt, NULL);
-	offset = odp_packet_l4_offset(pkt);
-
-	if (offset == ODP_PACKET_OFFSET_INVALID || ip == NULL)
-		return TEST_SEQ_INVALID;
-
-	if (ip->proto == ODPH_IPPROTO_IGMP) {
-		odp_packet_copy_to_mem(pkt, offset + ODP_IGMP_HLEN,
-				       sizeof(data), &data);
-
-	} else if (ip->proto == ODPH_IPPROTO_ICMPV4) {
-		odp_packet_copy_to_mem(pkt, offset + ODPH_ICMPHDR_LEN,
-				       sizeof(data), &data);
-	} else if (ip->proto == ODPH_IPPROTO_SCTP) {
-		odp_packet_copy_to_mem(pkt, offset + ODPH_SCTPHDR_LEN,
-				       sizeof(data), &data);
-	} else if (ip->proto == ODPH_IPPROTO_UDP) {
-		udp = (odph_udphdr_t *)odp_packet_l4_ptr(pkt, NULL);
-		port = odp_be_to_cpu_16(udp->dst_port);
-		if (port == ODP_GTPU_UDP_PORT) {
-			hlen = offset + ODPH_UDPHDR_LEN + ODP_GTP_HLEN;
-			odp_packet_copy_to_mem(pkt, hlen, sizeof(data), &data);
-		} else {
-			odp_packet_copy_to_mem(pkt, offset + ODPH_UDPHDR_LEN,
-					       sizeof(data), &data);
-		}
-	} else {
-		tcp = (odph_tcphdr_t *)odp_packet_l4_ptr(pkt, NULL);
-		odp_packet_copy_to_mem(pkt, offset + tcp->hl * 4,
-				       sizeof(data), &data);
-	}
+	uint32_t offset;
+
+	offset = seqno_offset(pkt);
+
+	odp_packet_copy_to_mem(pkt, offset, sizeof(data), &data);
```

Comment:
I think the success of the copy_to_mem operation should be checked. If it fails, the magic number may still match because the now uninitialized data may still hold the magic number from a previous invocation.

Reply:
Ack